src/states/node.rs

```diff
@@ -3418,6 +3418,13 @@ impl Node {
         self.clients_rate_limiter.get_clients_usage()
     }
 
+    pub fn get_banned_client_ips(&self) -> BTreeSet<IpAddr> {
+        self.banned_client_ips
+            .peek_iter()
+            .map(|(ip, _)| *ip)
+            .collect()
```

Comment:
(Not meant as a review for this PR) Wouldn't it be better if lru-time-cache (our crate) implemented `keys()` function ?